### ISSUE TITLE
How to allow the upload and editing of a XML text in Knora ?

### ISSUE THREAD

**Issue by mrivoal** [dhlab-basel/Knora#1455]
After a discussion with @lrosenth and @subotic, here is an issue based on a use case we have : a project would like to submit and update a text (TEI-XML) in Knora using a graphic interface. 

The corpus: 1000 to 1200 texts, ranging from one paragraph to 40 pages, with 10-15 tags

The **workflow**: 
- the project provides a standoff mapping for her TEI-XML edition 
- users uses Oxygen to generate the TEI-XML editions
- by copy-pasting the TEI-XML text from Oxygen, they will upload the TEI-XML text in a dedicated text field in Knora (and Salsah/knora-app)
- if the TEi-XML text uploaded needs to be edited, they will edit it in Oxygen and copy-paste the new version of the text in the text field in Knora

**What we think we need in Knora**:
- a new `salsah-gui:guielement`, similar to `Richtext`, but not enforcing the choice of the standard mapping
- we could possibly use the `salsah-gui-guiAttribute`property to link a `TextValue` property to the right standoff mapping, such as:

```
### http://www.knora.org/ontology/0113/lumieres-lausanne#transcriptionHasText

lumieres-lausanne:transcriptionHasText rdf:type owl:ObjectProperty ;
                    rdfs:subPropertyOf knora-base:hasValue ;
                    rdfs:label """Texte"""@fr ;
                    rdfs:comment """Texte de la transcription diplomatique"""@fr ;
                    knora-base:subjectClassConstraint lumieres-lausanne:Transcription ;
                    knora-base:objectClassConstraint knora-base:TextValue ;
                    salsah-gui:guiElement salsah-gui:Richtext ,
                    salsah-gui:guiAttribute "standoff-mapping=<http://rdfh.ch/projects/0113/mappings/manuscript-mapping>" .

```

**Comment by benjamingeer** [comment]
I think you'll need to discuss this with @tobiasschweizer, who is on holiday, so please be patient.

**Comment by tobiasschweizer** [comment]
Does this mainly affect SALSAH 1.5?

**Comment by mrivoal** [comment]
Since there is no alternative to edit data at the moment, yes. 

But since it make take some time until there is a viable standoff editor in Knora-app, I think it could/should also affect Knora-app.

**Comment by tobiasschweizer** [comment]
>  I think it could/should also affect Knora-app.

Are you solely referring to the GUI or do you think there are requirements affecting the Knora API?

**Comment by mrivoal** [comment]
I am referring to the GUI. I couldn't say if some of these requirements would affect the current Knora API.

This possibly will?
> we could possibly use the `salsah-gui-guiAttributeproperty` to link a `TextValue` property to the right standoff mapping

**Comment by tobiasschweizer** [comment]
If you create or modify a `TextValue`, a mapping has to be indicated. In the simplest case, an existing mapping can be used. Otherwise a new mapping has to be created first and then be indicated.

**Comment by mrivoal** [comment]
Yes, the idea here is that they would indicate the right (project-specific) mapping (which won't be the standard mapping) in the ontology itself for a given `TextValue`.

**Comment by flavens** [comment]
issue moved to DSP-1219
